ONSdigital/dp-dataset-api
# Add new health check and update app dependencies

### What

Add new health check endpoint `/health` with several checkers:
* Kafka Producer (event)
* Kafka Audit Producer
* Mongo db
* Graph db (neo4j)
* Zebedee

Also did the following:
* Use go modules instead of go vendor
* Update ci scripts and makefile fo go modules
* Use new kafka, mongo, log and api-client library
* Update graceful shutdown
* Update documentation
* Update nomad plan for publishing and web `/health` endpoint
* Update all logs, pass context to methods with logs and update unit tests

### How to review

* Check sanity of code
* Check unit tests pass
* Check healthcheck works in web and publishing modes, use `curl localhost:22000/health`
* Check endpoints return the expected json (matches swagger spec)

### Who can review

Anyone
